PYME/LMVis/pipeline.py

```diff
@@ -767,64 +780,49 @@ def OpenFile(self, filename= '', ds = None, **kwargs):
-            mapped_ds.addVariable('pixelSize', kwargs['PixelSize'])
-            mapped_ds.setMapping('x', 'x*pixelSize')
-            mapped_ds.setMapping('y', 'y*pixelSize')
-
-        #extract information from any events
-        self.ev_mappings, self.eventCharts = _processEvents(mapped_ds, self.events, self.mdh)
+        # skip the MappingFilter wrapping, etc. in self.addDataSource and add this datasource as-is
+        self.dataSources['FitResults'] = ds
 
+        add_pipeline_variables = Pipelineify(self.recipe, 
+            inputFitResults='FitResults',
+            pixelSizeNM=1. if 'PixelSize' not in kwargs.keys() else kwargs['PixelSize'],
+            outputLocalizations='Localizations')
+        self.recipe.add_module(add_pipeline_variables)
 
+        # FIXME - we do this already in pipelinify, maybe we can avoid doubling up?
+        self.ev_mappings, self.eventCharts = _processEvents(ds, self.events, self.mdh)  # extract information from any events
 
         #Fit module specific filter settings        
         if 'Analysis.FitModule' in self.mdh.getEntryNames():
             fitModule = self.mdh['Analysis.FitModule']
-            
-            #print 'fitModule = %s' % fitModule
-            
             if 'Interp' in fitModule:
                 self.filterKeys['A'] = (5, 100000)
-            
-            if 'LatGaussFitFR' in fitModule:
-                mapped_ds.addColumn('nPhotons', getPhotonNums(mapped_ds, self.mdh))
-
-            if 'SplitterFitFNR' in fitModule:
-                mapped_ds.addColumn('nPhotonsg', getPhotonNums({'A': mapped_ds['fitResults_Ag'], 'sig': mapped_ds['fitResults_sigma']}, self.mdh))
-                mapped_ds.addColumn('nPhotonsr', getPhotonNums({'A': mapped_ds['fitResults_Ar'], 'sig': mapped_ds['fitResults_sigma']}, self.mdh))
-                mapped_ds.setMapping('nPhotons', 'nPhotonsg+nPhotonsr')
-
             if fitModule == 'SplitterShiftEstFR':
                 self.filterKeys['fitError_dx'] = (0,10)
                 self.filterKeys['fitError_dy'] = (0,10)
 
         #self._get_dye_ratios_from_metadata()
 
-        self.addDataSource('Localizations', mapped_ds)
-
         # Retrieve or estimate image bounds
-        if False:  # 'imgBounds' in kwargs.keys():
-            self.imageBounds = kwargs['imgBounds']
-        elif (not (
-                'scanx' in mapped_ds.keys() or 'scany' in mapped_ds.keys())) and 'Camera.ROIWidth' in self.mdh.getEntryNames():
+        if False:  # 'imgBounds' in kwargs.keys(): 
+            # TODO - why is this disabled? Current usage would appear to be when opening from LMAnalysis
+            # during real-time localization, to force image bounds to match raw data, but also potentially useful
+            # for other scenarios where metadata is not fully present.
+             self.imageBounds = kwargs['imgBounds']
+        elif ('scanx' not in ds.keys() or 'scany' not in ds.keys()) and 'Camera.ROIWidth' in self.mdh.getEntryNames():
```

Comment:
This needs to run on the **mapped** data, not the raw data (both the test for keys and `ImageBounds.estimateFromSource`.

Reply:
Hacky and mostly equivalent solution would be is to estimate `self.imageBounds` based on `self.dataSources['Localisations']` *after* the call to `recipe.execute()`, need to think if this (a) makes sense, or whether we should b) use `.selectedDataSource` instead, or c) do the bounds estimation in `Pipelineify` instead.